src/main/java/com/cofancs/SortingAlgorithm.java

```diff
@@ -7,44 +7,49 @@
     public static final String INPUT_ARRAY_SHOULDN_T_BE_NULL = "inputArray shouldn't be null";
     public static final String INPUT_CAN_CONTAINS_DIGITS_AND_A_SINGLE_CHARACTERS = "Input can contains digits and a single \"-\" characters! ";
     public static final String DASH = "-";
+    public static final Character DASH_CHAR = '-';
 
-    public  String[] sortNullsToTheEnd(String[] inputArray){
+    private static boolean testIfContainsInvalidObjects(Object object) {
+        return !((object == null) || (object instanceof Integer) || ((object instanceof Character) && object.equals(DASH_CHAR)) || ((object instanceof String) && object.equals(DASH)));
+    }
+
+    public Object[] sortNullsToTheEnd(Object[] inputArray) throws IllegalArgumentException {
         checkCorrectInput(inputArray);
-        String[] result =doSortingTheEmptyCharacters(inputArray);
+        Object[] result = doSortingTheEmptyCharacters(inputArray);
         return result;
     }
 
-    private String[] doSortingTheEmptyCharacters(String[] inputArray) {
-        for(int i=0;i<inputArray.length-1;i++){
-            if(inputArray[i].equals(DASH)){
-                String element=inputArray[i];
-                for(int j = i+1; j<inputArray.length;j++){
-                    if(!inputArray[j].equals(DASH)){
-                        inputArray[i]=inputArray[j];
-                        inputArray[j]=element;
+    private Object[] doSortingTheEmptyCharacters(Object[] inputArray) {
+        for (int i = 0; i < inputArray.length; i++) {
+            if (inputArray[i] == null || inputArray[i].equals(DASH) || inputArray[i].equals(DASH_CHAR)) {
```

Comment:
Extract the check to a private function: isEmptyValue() and reuse where possible